Libraries/plist/Tools/PlistBuddy.cpp

```diff
@@ -172,45 +172,20 @@ CommandHelp()
 #undef INDENT
 }
 
-static std::unique_ptr<plist::Object>
-ReadPropertyList(Filesystem const *filesystem, std::string const &path)
-{
-    if (path.empty()) {
-        fprintf(stderr, "error: no input specified\n");
-        return NULL;
-    }
-
-    if (!filesystem->exists(path)) {
-        fprintf(stderr, "error: file does not exist: %s\n", path.c_str());
-        return NULL;
-    }
-
-    std::vector<uint8_t> contents;
-    if (!filesystem->read(&contents, path)) {
-        fprintf(stderr, "error: unable to read %s\n", path.c_str());
-        return NULL;
-    }
-
-    auto deserialize = plist::Format::Any::Deserialize(contents);
-    if (!deserialize.first) {
-        fprintf(stderr, "error: %s\n", deserialize.second.c_str());
-        return NULL;
-    }
-
-    return std::move(deserialize.first);
-}
-
 static void
 ParseCommandKeyPathString(std::string const &keyPathString, std::queue<std::string> *keyPath)
 {
     std::string::size_type prev = 0;
     do {
         std::string::size_type pos = keyPathString.find(':', prev);
         std::string::size_type end = (pos == std::string::npos ? pos : pos - prev);
+        bool emptyInitialKey = (prev == 0 && pos == 0);
 
-        std::string key = keyPathString.substr(prev, end);
-        if (!key.empty()) {
-            keyPath->push(key);
+        if (!emptyInitialKey) {
```

Comment:
In `PlistBuddy` key specifiers can be `:CFBundleVersion` or just `CFBundleVersion`; i.e. the initial colon is optional. If it's there, then the contents up to the first `:` is empty, so don't pass that empty string along as a key name.